### PR TITLE
Expose constraints in tooltips

### PR DESCRIPTION
**Pull request by elpaso**
immagine

### CODE REVIEW

File: src/core/browser/qgsfieldsitem.cpp [hunk 1]
```diff
@@ -148,6 +148,20 @@ QgsFieldItem::QgsFieldItem( QgsDataItem *parent, const QgsField &field )
   // Precondition
   Q_ASSERT( static_cast<QgsFieldsItem *>( parent ) );
   setState( Qgis::BrowserItemState::Populated );
+  const auto constraints { field.constraints().constraints() };
+  QStringList constraintsText;
+  if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintNotNull ) )
+  {
+    constraintsText.push_back( tr( "Not Null" ) );
```
Comment (m-kuhn) [inline src/core/browser/qgsfieldsitem.cpp:155]: I think that capitalization is still missing, we currently have:

 - Not null
 - not null
 - NOT NULL
 - Not NULL
 
 ```
 grep -i 'Not NULL' qgis_de.ts | grep source
        <source>Name of raster map to use for masking. Only cells that are not NULL and not zero are interpolated</source>
        <source>Is not missing (not null)</source>
        <source>is not null</source>
        <source>Not null</source>
        <source>&lt;p&gt;Enforcing the not null constraint prevents committing features which do not meet the constraint.&lt;/p&gt;&lt;p&gt;Unenforced constraints display a warning to users, but do not prevent committing the feature.&lt;/p&gt;</source>
        <source>Enforce not null constraint</source>
        <source>The provider for this layer has a NOT NULL constraint set on the field.</source>
        <source>Not NULL</source>
        <source>0, if your geometry is not NULL</source>
        <source>To test for NULL use an &lt;i&gt;IS NULL&lt;/i&gt; or &lt;i&gt;IS NOT NULL&lt;/i&gt; expression.</source>
        <source>Not null</source>
        <source>Unique column &apos;%1&apos; doesn&apos;t have a NOT NULL constraint.</source>
        <source>Unique column &apos;%1&apos; doesn&apos;t have a NOT NULL constraint.</source>
        <source>is not null</source>
```

Jokes aside, should we settle on one?
I'd go all caps, but no strong opinion.

Comment (elpaso) [reply src/core/browser/qgsfieldsitem.cpp:155]: Yeah, all caps works for me too.

Comment (elpaso) [reply src/core/browser/qgsfieldsitem.cpp:155]:
```suggestion
    constraintsText.push_back( tr( "NOT NULL" ) );
```

File: src/core/browser/qgsfieldsitem.cpp [hunk 2]
```diff
@@ -148,6 +148,20 @@ QgsFieldItem::QgsFieldItem( QgsDataItem *parent, const QgsField &field )
...
+  }
+  if ( constraints.testFlag( QgsFieldConstraints::Constraint::ConstraintUnique ) )
+  {
+    constraintsText.push_back( tr( "Unique" ) );
```
Comment (elpaso) [inline src/core/browser/qgsfieldsitem.cpp:159]:
```suggestion
    constraintsText.push_back( tr( "UNIQUE" ) );
```

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**